include/geode/geometry/basic_objects/cylinder.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2019 - 2023 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#pragma once
+
+#include <geode/geometry/basic_objects/segment.h>
+#include <geode/geometry/common.h>
+
+namespace geode
+{
+    using Basis3D = std::array< Vector3D, 3 >;
+    class Cylinder
+    {
+    public:
+        Cylinder( const Segment3D& axis, double radius );
+
+        Cylinder( const Cylinder& other );
+        Cylinder& operator=( const Cylinder& other );
+        Cylinder( Cylinder&& other );
+        Cylinder& operator=( Cylinder&& other );
+
+        const Segment3D& axis() const;
+        const Basis3D& basis() const;
+        double radius() const;
+
+    private:
+        Segment3D axis_;
+        double radius_{ 0 };
+        Basis3D basis_;
```

Comment:
Should this really be a class object, or should it be computed only when needed (and eventually stored in a CacheData) ?

Reply:
Good question, any thought @BotellaA?

Reply:
Is it really Cylinder related or Segment/Cylinder intersection related? maybe move it in intersection.cpp?